mainnet-contracts/src/WithdrawalManager.sol

```diff
@@ -121,37 +108,46 @@ contract WithdrawalManager {
             PUFFER_VAULT.transferETH(address(this), transferAmount);
             PUFFER_VAULT.burn(pufETHBurnAmount);
 
-            batch.pufETHToEthExchangeRate = batchFinalizationExchangeRate;
+            batch.pufETHToEthExchangeRate = uint64(batchFinalizationExchangeRate);
 
             emit BatchFinalized(i, expectedETHAmount, transferAmount, pufETHBurnAmount);
 
             unchecked {
                 ++i;
             }
         }
-        finalizedWithdrawalBatch = withdrawalBatchIndex;
+        $.finalizedWithdrawalBatch = withdrawalBatchIndex;
     }
 
     /**
-     * @dev Complete a queued withdrawal
-     * @param withdrawalIdx The index of the withdrawal to complete
+     * @inheritdoc IWithdrawalManager
+     * @dev Restricted in this context is like `whenNotPaused` modifier from Pausable.sol
      */
-    function completeQueuedWithdrawal(uint256 withdrawalIdx) external {
-        if (withdrawalIdx < finalizedWithdrawalBatch * BATCH_SIZE) {
+    function completeQueuedWithdrawal(uint256 withdrawalIdx) external restricted {
+        WithdrawalManagerStorageStruct storage $ = _getWithdrawalManagerStorage();
+
+        if (withdrawalIdx < $.finalizedWithdrawalBatch * BATCH_SIZE) {
             revert NotFinalized();
         }
 
-        Withdrawal memory withdrawal = withdrawals[withdrawalIdx];
-        uint256 batchSettlementExchangeRate = withdrawalBatches[withdrawalIdx / BATCH_SIZE].pufETHToEthExchangeRate;
+        Withdrawal memory withdrawal = $.withdrawals[withdrawalIdx];
+        uint256 batchSettlementExchangeRate = $.withdrawalBatches[withdrawalIdx / BATCH_SIZE].pufETHToEthExchangeRate;
 
         uint256 payoutExchangeRate = Math.min(withdrawal.pufETHToEthExchangeRate, batchSettlementExchangeRate);
         uint256 payoutAmount = (uint256(withdrawal.pufETHAmount) * payoutExchangeRate) / 1 ether;
 
         // remove data for some gas savings
-        delete withdrawals[withdrawalIdx];
+        delete $.withdrawals[withdrawalIdx];
 
         emit WithdrawalCompleted(withdrawalIdx, payoutAmount, payoutExchangeRate, withdrawal.recipient);
 
         Address.sendValue(payable(withdrawal.recipient), payoutAmount);
```

Comment:
Do we care if the recipient is a smart contract that doesn't have payable recieve|fallback?
It is the user's fault, but still.. do we want to wrap it into WETH & transfer?